src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/equalsavoidnull/Example1.java

```diff
@@ -0,0 +1,23 @@
+/*xml
+<module name="Checker">
+  <module name="TreeWalker">
+    <module name="EqualsAvoidNull"/>
+  </module>
+</module>
+*/
+
+package com.puppycrawl.tools.checkstyle.checks.coding.equalsavoidnull;
+
+// xdoc section -- start
+public class Example1 {
+  public void foo() {
+    String nullString = null;
+    // violation below 'String literal expressions should be on the left side of an equals comparison'
+    nullString.equals("My_Sweet_String");
+    "My_Sweet_String".equals(nullString);
+    // violation below 'String literal expressions should be on the left side of an equalsIgnoreCase comparison'
```

Comment:
To avoid line length violation try
`// violation below 'String literal expressions should be on the left side'`

It is descriptive enough to let human understand how it works.

Reply:
Okay, I will make the change